device/sycl/src/clusterization/clusterization_algorithm.cpp

```diff
@@ -55,93 +54,117 @@ clusterization_algorithm::output_type clusterization_algorithm::operator()(
 
     // Helper container for sparse CCL calculations
     vecmem::data::jagged_vector_buffer<unsigned int> sparse_ccl_indices(
-        cell_sizes_plus, m_mr.get());
+        cell_sizes_plus, (m_mr.device ? *(m_mr.device) : m_mr.host),
+        (m_mr.device ? &(m_mr.host) : nullptr));
```

Comment:
:thinking: Right... I forgot about this...

The logic in vecmem is just the opposite to how I made you set up `traccc::memory_resource`. Jagged buffers (amongst other things) assume that the "main" memory resource may or may not be host accessible. And if its not, then the optional `vecmem::memory_resource` pointer would provide the host-accessible resource.

Instead of implementing this reverse logic everywhere, I think we should rather make `traccc::memory_resource` follow vecmem's logic. I.e. implement it like:

```c++
namespace traccc {
    struct memory_resource {
        vecmem::memory_resource& main;
        vecmem::memory_resource* host = nullptr;
    };
}
```

This way we could feed those variables directly into the buffer constructors... :thinking:

Reply:
done